lexibank_lieberherrkhobwa.py

```diff
@@ -1,106 +1,131 @@
                     Name=concept.english,
                     Concepticon_ID=concept.concepticon_id,
                     Concepticon_Gloss=concept.concepticon_gloss,
                 )
 
-            # Read raw languages and append it
-            langs = {}
-            for language in self.languages:
-                langs[language["Name"]] = language["Source"]
-                ds.add_language(
-                    ID=language["Name"],
-                    Name=language["Name"],
-                    Glottocode=language["Glottocode"],
-                    Glottolog_Name=language["Glottolog_Name"],
-                )
+            # Read raw languages and add to the dataset; at the same time,
+            # build a map that can be used for iterating over language ids
+            langs = {l['Name']:l['Source'] for l in self.languages}
+            ds.add_languages()
 
             # iterate over the source adding lexemes and collecting cognates
             for row in data[2:]:
+                # Get the language_id, which also allows to skip over
+                # data we don't want
                 lid = slug(row[0])
+
                 if lid in langs:
                     for cid in range(1, len(row), 2):
-                        form = row[cid]
-                        if not form or (form == "NA"):
+                        # Skip over rows with empty fields for cogid
+                        if not row[cid+1]:
+                            continue
+                            
+                        # Compute a cognate_id number; lingpy now requires
+                        # this to be an integer
+                        cognate_id = cid * 100 + int(row[cid + 1])
+                    
+                        # Extract the value from the raw data, skipping over
+                        # missing or non-existing forms
+                        value = row[cid].strip()
+                        if not value or value == "NA":
                             continue
-                        ipa = form
-                        ipa = ipa.replace("- ", "-").replace(" ", "_")
-                        ipa = strip_brackets(ipa).strip()
-
-                        if ipa:
-                            # split ipa and remove final/initial underscores if any
-                            # (without regular expressions)
-                            ipa = split_text(ipa, "~/~⪤,")[0]
-
-                            if ipa[0] == "_":
-                                ipa = ipa[1:].strip()
-                            if ipa[-1] == "_":
-                                ipa = ipa[:-1].strip()
-
-                            # lingpy now requires this to be an integer
-                            cognate_id = cid * 100 + int(row[cid + 1])
 
-                            for lex in ds.add_lexemes(
+                        # Build the form from the value, splitting the text
+                        # first
+                        form = split_text(value, "~/~⪤,")[0]
```

Comment:
What you do now is: you create a non-splitting dataset, then you split the forms, and pass them as value. This means, you essentially corrupt the original data, as you do not have any value that is the "true" value.

Reply:
It is not a non-splitting dataset. It is emitting only the first form of each entry, thus matching what was in the previous version, but I would prefer to emit all the forms (as in my last PR message). This is easy to change.

I am not sure I understand the "you do not have any value that is the 'true' value", though: entries with no separators to split (to examplify with a singleton, `tʰa-rija-ma-ge`) are reported as in the raw data. You mean having the non-splitted raw value in the same row of each form coming from that value?

Reply:
Check these previous lines in forms.csv:

```
 rahung-Lieberherr-2017-100-3-1,,rahung,Lieberherr-2017-100-3,hɛ ~ ʔɛ,hɛ,h ɛ,,Lieberherr2017,501, 
 rahung-Lieberherr-2017-100-3-2,,rahung,Lieberherr-2017-100-3,hɛ ~ ʔɛ,ʔɛ,ʔ ɛ,,Lieberherr2017,501,  
```
and the other new line:

```
 rahung-Lieberherr-2017-100-3-1,,rahung,Lieberherr-2017-100-3,hɛ ,hɛ,h ɛ,,Lieberherr2017,501, 
```
Your *value* is `hɛ `. But it shoudl be `hɛ ~ ʔɛ`. 

You split the value into several forms, take the first one only, and then pass that *form* as *value*. This is of course wrong. Value is what we find in the cell in the data, form is what we process.

Reply:
You are allowed to skip the `ʔɛ`, if you are convinced, the forms in the cells are just variants. But then you should use (current syntax) a non-splitting dataset, but NOT preprocess the forms outside the class functions of cldf.dataset. But you still keep doing that. So keep the NonSplittingDataset if you believe it is okay, but make sure the splitting is then corrected for, and don't split in the loop.

Reply:
Ah, of course, got it. :+1+